apps/roboshield/src/components/RichText/styleConverter.tsx

```diff
@@ -0,0 +1,48 @@
+import { JSXConverters } from "@payloadcms/richtext-lexical/react";
+import { RichTypography, Link } from "@commons-ui/next";
+
+const DEFAULT_PROPS = {
+  html: false,
+};
+
+export const styleConverter = (converterProps: any): JSXConverters<any> => ({
+  heading: ({ node, nodesToJSX }) => {
+    const Tag = node.tag || "h1";
+    return (
+      <RichTypography
+        variant={Tag}
+        component={Tag}
+        {...converterProps.typographyProps}
+      >
+        {nodesToJSX({ nodes: node.children })}
+      </RichTypography>
+    );
+  },
+  quote: ({ node, nodesToJSX }) => (
+    <blockquote>{nodesToJSX({ nodes: node.children })}</blockquote>
+  ),
+  link: ({ node, nodesToJSX }) => {
+    return (
+      <RichTypography
+        component={Link}
+        href={node.fields.url || node.href || ""}
+        {...node?.typographyProps}
+        {...converterProps.typographyProps}
+        {...DEFAULT_PROPS}
+      >
+        {nodesToJSX({ nodes: node.children })}
+      </RichTypography>
+    );
+  },
+  paragraph: ({ node, nodesToJSX }) => {
+    return (
+      <RichTypography
+        {...node?.typographyProps}
+        {...DEFAULT_PROPS}
+        {...converterProps.typographyProps}
```

Comment:
We can create:
```jsx
const props = { ...node?.typographyProps, ...DEFAULT_PROPS, ...converterProps.typographyProps };
```
Somewhere at the top of the function instead of repeating them on each match.

PS: Is it `node`, `DEFAULT` and then `converterProps` **or** `node`, `converterProps` and then `DEFAULT`?

Reply:
I think as we increase the order of specificity it should be `DEFAULT`, `converterProps`, `node`